src/placeholder/solvers/interface.py

```diff
@@ -0,0 +1,101 @@
+
+class ModelNotDefinedError(Exception):
+    pass
+
+
+class SolverNotDefinedError(Exception):
+    pass
+
+
+class Solver:
+
+    def __init__(self, model_instance=None):
+        self._model = model_instance
+        self.success = None
+        self.x_opt = None
+        self.fun_val_opt = None
+        self.options = None
+        self.success = None
+
+    @property
+    def model(self):
+        return self._model
+
+    @model.setter
+    def model(self, model_instance):
+        self._model = model_instance
+
+    def assert_models_defined(self):
+        if self._model is not None:
+            return True
+        else:
+            raise ModelNotDefinedError()
+    
+    def set_options(self, options):
+        self.options = options
+
+    def fit(self, data, x_init=None, options=None):
+        raise NotImplementedError()
+
+    def predict(self, **kwargs):
+        raise NotImplementedError()
+
+
+class CompositeSolver(Solver):
+
+    def __init__(self, solvers_list=None):
+        super().__init__(model_instance=None)
+        if solvers_list is not None:
+            self._solvers = solvers_list
+        else:
+            self._solvers = []
+
+    @property
+    def solvers(self):
+        return self._solvers
+
+    @solvers.setter
+    def solvers(self, solvers_list):
+        self._solvers = solvers_list
+
+    def add_solver(self, solver):
+        self._solvers.extend(solver)
+
+    def set_options(self, options):
+        if self.assert_solvers_defined():
+            for solver in self._solvers:
+                solver.set_options(options)
+
+    @property
+    def model(self):
+        models = []
+        if self.assert_solvers_defined() is True:
+            for solver in self._solvers:
+                models.append(solver.model)
+        return models
+
+    @model.setter
+    def model(self, model_instances):
+        if isinstance(model_instances, list):
+            assert len(model_instances) == len(self._solvers)
```

Comment:
@jizezhang I've been making explicit subclasses of `Exception`, see some in `data.py`. Might be over-engineered but I think it will be helpful in the future if we have them well-defined.